File: frontend/interfaces/product.ts
export default interface IProduct {
    readonly id: string;
    readonly product_id: string;
    readonly name: string;
    readonly sku: string;
    readonly description: string;
    readonly price: number;
    readonly sale_price: number;
    readonly regular_price: number;
    readonly discountBegin: Date;
    readonly discountEnd: Date;
    readonly category: string;
    readonly quantity: number;
    readonly images: string[];
    readonly width?: number;
    readonly weight?: number;
    readonly length?: number;
    readonly height?: number;
    readonly countryOrigin?: string;
    readonly tags?: string[];
    readonly categoryRaw?: any;
    readonly expiryDate?: Date;
  }
  

File: havafy-api/models/product.go
package models

import "gorm.io/gorm"

//Product model
type Product struct {
	gorm.Model

	Name  string `json:"name"`
	Description string `json:"description"`
	Price float64 `json:"description"`
	Quantity int `json:"quantity"`
	SalePrice float64 `json:"sale_price"`
	Visibility string `json:"visibility"`
	Thumbnail string `json:"thumbnail"`
	Images string `json:"images"`

}

// Products struct
type Products struct {
	Products []Product `json: "products"`
}